pkg/store/flags_test.go

```diff
@@ -9,6 +9,68 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestHasPriority(t *testing.T) {
+	tests := []struct {
+		name         string
+		currentState *Flags
+		storedSource string
+		newSource    string
+		hasPriority  bool
+	}{
+		{
+			name:         "same source",
+			currentState: &Flags{},
+			storedSource: "A",
+			newSource:    "A",
+			hasPriority:  true,
+		},
+		{
+			name: "no priority",
+			currentState: &Flags{
+				FlagSources: []string{
+					"B",
+					"A",
+				},
+			},
+			storedSource: "A",
+			newSource:    "B",
+			hasPriority:  false,
+		},
+		{
+			name: "priority",
+			currentState: &Flags{
+				FlagSources: []string{
+					"A",
+					"B",
+				},
+			},
+			storedSource: "B",
+			newSource:    "A",
+			hasPriority:  false,
+		},
```

Comment:
```suggestion
		{
			name: "priority",
			currentState: &Flags{
				FlagSources: []string{
					"A",
					"B",
				},
			},
			storedSource: "A",
			newSource:    "B",
			hasPriority:  true,
		},
```

I think this test is a lie. I believe it should be the above, which then covers an additional and important (I think?) code path.

The coverage seems to back this up, it goes from `87.9% of statements` to `89.0% of statements` for this package with this change.